ras/hwinfo.py

```diff
@@ -71,8 +72,9 @@ def test(self):
         self.run_cmd("hwinfo --debug 0 --disk --log=-")
         self.run_cmd("hwinfo --short --block")
         self.run_cmd("hwinfo --disk --save-config=all")
-        if "failed" in process.system_output("hwinfo --disk --save-config=all | "
-                                             "grep failed | tail -1", shell=True):
+        if "failed" in process.system_output("hwinfo --disk --save-config"
+                                             "=all | ""grep failed | tail -1",
+                                             shell=True):
```

Comment:
@PraveenPenguin can you change it into native python parsing please ?

Reply:
sure make sense will do

Reply:
previously the testcase checks for last message logged into dmesg for hwinfo errors, but with the current change there might be false positives because the check for `failed` is for the whole dmesg those might not be from `hwinfo`. You probably want to clear the `dmesg` before running the test.